### ISSUE TITLE
Error using Redis with template cache

### ISSUE THREAD

**Issue by markbaird** [pencilblue/pencilblue#611]
If you setup Redis (not fakeredis) as your system's cache, and set cache templates to true like this:
```
        "templates": {
            "use_memory": true,
            "use_cache": true
        },
```
Then your site behaves as follows:
* Load homepage once and everything is fine, loading your installed template
* Refresh the page and everything is suddenly using the default PencilBlue template
* Refresh the page a second time and you get this error:
```
2015-05-16T03:09:52.825Z - silly: [M] SettingService1: Checking Service [Memory] for Key [active_theme]
2015-05-16T03:09:52.825Z - silly: [M] TemplateService: Checking Service [Memory] for Key [d:\Projects\blog\plugins\codeammo-theme\templates\elements\media.html]
2015-05-16T03:09:52.825Z - silly: [M] TemplateService: Checking Service [Cache] for Key [d:\Projects\blog\plugins\codeammo-theme\templates\elements\media.html]
2015-05-16T03:09:52.826Z - silly: [M] TemplateService: Checking Service [Template] for Key [d:\Projects\blog\plugins\codeammo-theme\templates\elements\media.html]
2015-05-16T03:09:52.826Z - silly: [M] TemplateEntityService: Attempted to retrieve template content.  FILE=[d:\Projects\blog\plugins\codeammo-theme\templates\elements\media.html] CONTENT_LEN=[n/a]
2015-05-16T03:09:52.826Z - silly: [M] TemplateService: Checking Service [Memory] for Key [d:\Projects\blog\plugins\codeammo-theme\templates\elements\media.html]
2015-05-16T03:09:52.826Z - silly: [M] TemplateService: Checking Service [Cache] for Key [d:\Projects\blog\plugins\codeammo-theme\templates\elements\media.html]
2015-05-16T03:09:52.827Z - silly: [M] TemplateService: Checking Service [Template] for Key [d:\Projects\blog\plugins\codeammo-theme\templates\elements\media.html]
2015-05-16T03:09:52.827Z - silly: [M] TemplateEntityService: Attempted to retrieve template content.  FILE=[d:\Projects\blog\plugins\codeammo-theme\templates\elements\media.html] CONTENT_LEN=[n/a]
2015-05-16T03:09:52.827Z - silly: [M] TemplateService: Checking Service [Memory] for Key [d:\Projects\blog\plugins\pencilblue\templates\elements\media.html]
2015-05-16T03:09:52.830Z - error: [M] RequestHandler: An error occurred during controller execution. URL=[GET:/] ROUTE={"path":"","pattern":"^/[/]{0,1}$","path_vars":{},"expression":{},"themes":{"pencilblue":{"GET":{"method":"GET","path":"/","access_level":0,"auth_required":false,"content_type":"text/html"}},"codeammo-theme":{"GET":{"method":"GET","path":"/","auth_required":false,"content_type":"text/html"}}}}
Error: TemplateService: A valid content string is required in order for the template engine to process the value. Content='[object Object]'
    at TemplateService.process (d:\Projects\blog\include\service\entities\template_service.js:306:16)
    at d:\Projects\blog\include\service\entities\template_service.js:292:18
    at d:\Projects\blog\include\service\entities\template_service.js:266:21
    at Object.async.whilst (d:\Projects\blog\node_modules\async\lib\async.js:686:13)
    at d:\Projects\blog\node_modules\async\lib\async.js:682:23
    at d:\Projects\blog\include\service\entities\template_service.js:262:25
    at d:\Projects\blog\include\service\simple_layered_service.js:95:17
    at Object.async.whilst (d:\Projects\blog\node_modules\async\lib\async.js:686:13)
    at d:\Projects\blog\node_modules\async\lib\async.js:682:23
    at d:\Projects\blog\include\service\simple_layered_service.js:76:21
```
From that point your site will always throw that error unless you disable template cache. If you clear the Redis cache then you will get the 3 step process again like I described above.

**Comment by brianhyder** [comment]
Confirmed as a bug.  The issue is that the content isn't getting correctly transformed back into an object coming from cache.  A vast majority of the time you wouldn't need in-memory and cache which is why it hasn't shown up until now.  Good catch.

**Comment by markbaird** [comment]
Yeah I didn't actually intend to use Redis for the template cache, I just enabled it by accident when playing with the cache settings.

**Comment by brianhyder** [comment]
This should now be fixed as part of PR #620
